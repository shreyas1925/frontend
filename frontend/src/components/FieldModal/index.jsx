import React from 'react';
import './FieldModal.css';
import axios from 'axios';
import PropTypes from 'prop-types';

export default function FieldModal(props) {
    const [field, setField] = React.useState('');

    if (!props.show) {
        return null;
    }

    const handleChange = (e) => {
        setField(e.target.value);
    };

    const addField = async() => {
        const res = await axios({
            method:'POST',
            url:`http://localhost:5000/api/addFields/${props.contentId}`,
            data:{newfield:field}
        });

        console.log(res);
        window.location.reload();
    };


    return (
        <div className="modal-post" onClick={props.onClose}>
            <div className="modal-content-post" onClick={(e) => e.stopPropagation()}>
                <div className="modal-header-post">
                    <span className="modal-title-post">Create a new content type</span>
                </div>
                <div className="modal-body-post">
                    <span>Name of the content type</span>
                    <input type="text" name="content" onChange={handleChange} value={field} />
                </div>
                <div className="modal-footer-post">
                    <button onClick={props.onClose} className="modal-close-button-post">
            Close
                    </button>
                    <button className="modal-create-button-post" onClick={addField}>
                        Create
                    </button>
                </div>
            </div>
        </div>
    );
}

FieldModal.propTypes = {
    onClose: PropTypes.func.isRequired,
    show: PropTypes.bool.isRequired,
    contentId: PropTypes.string,
};