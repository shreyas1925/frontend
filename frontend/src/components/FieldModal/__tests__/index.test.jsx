import FieldModal from "..";
import { render } from "@testing-library/react";

describe("Field Modal", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<FieldModal show={true} />);
    expect(asFragment()).toMatchSnapshot();
  });


  describe("when the modal is open", () => {
    it("should render the modal", () => {
      const { getByText } = render(<FieldModal show={true} />);
      expect(getByText("Close")).toBeInTheDocument();
    });
  });
  describe("when the modal is closed", () => {
    it("should not render the modal", () => {
      const { queryByText } = render(<FieldModal show={false} />);
      expect(queryByText("Close")).not.toBeInTheDocument();
    });
  });
});
