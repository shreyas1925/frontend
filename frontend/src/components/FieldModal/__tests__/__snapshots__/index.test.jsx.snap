// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Modal should render correctly 1`] = `
<DocumentFragment>
  <div
    class="modal-post"
  >
    <div
      class="modal-content-post"
    >
      <div
        class="modal-header-post"
      >
        <span
          class="modal-title-post"
        >
          Create a new content type
        </span>
      </div>
      <div
        class="modal-body-post"
      >
        <span>
          Name of the content type
        </span>
        <input
          name="content"
          type="text"
          value=""
        />
      </div>
      <div
        class="modal-footer-post"
      >
        <button
          class="modal-close-button-post"
        >
          Close
        </button>
        <button
          class="modal-create-button-post"
        >
          Create
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
